### PR TITLE
feat: project view

### PR DESCRIPTION
**Pull request by fgerschwiler**

### CODE REVIEW

File: pkg/management/api/grpc/project_converter.go [hunk 1]
```diff
@@ -82,3 +164,89 @@ func projectRoleChangeToModel(role *ProjectRoleChange) *proj_model.ProjectRole {
...
+		return model.SEARCHMETHOD_STARTS_WITH
+	case SearchMethod_SEARCHMETHOD_EQUALS_IGNORE_CASE:
+		return model.SEARCHMETHOD_EQUALS_IGNORE_CASE
+	case SearchMethod_SEARCHMETHOD_CONTAINS_IGNORE_CASE:
+		return model.SEARCHMETHOD_CONTAINS_IGNORE_CASE
+	case SearchMethod_SEARCHMETHOD_STARTS_WITH_IGNORE_CASE:
+		return model.SEARCHMETHOD_STARTS_WITH_IGNORE_CASE
+	default:
+		return model.SEARCHMETHOD_EQUALS
+	}
```
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply pkg/management/api/grpc/project_converter.go:251]: i moved it to a separate file

File: internal/project/repository/view/model/application.go [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+		a.State = int32(model.APPSTATE_INACTIVE)
+	case es_model.ApplicationReactivated:
+		a.State = int32(model.APPSTATE_ACTIVE)
+	}
+	return nil
+}
+
+func (a *ApplicationView) setRootData(event *models.Event) {
+	a.ProjectID = event.AggregateID
+	a.ChangeDate = event.CreationDate
```
[the remark this answers is not in the document]
Comment (fgerschwiler) [reply internal/project/repository/view/model/application.go:146]: yes

File: internal/project/repository/view/application_view.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+func SearchApplications(db *gorm.DB, table string, req *proj_model.ApplicationSearchRequest) ([]*model.ApplicationView, int, error) {
+	apps := make([]*model.ApplicationView, 0)
+	query := view.PrepareSearchQuery(table, model.ApplicationSearchRequest{Limit: req.Limit, Offset: req.Offset, Queries: req.Queries})
+	count, err := query(db, &apps)
+	if err != nil {
+		return nil, 0, err
+	}
+	return apps, count, nil
```
Comment (adlerhurst) [inline internal/project/repository/view/application_view.go:24]:
```suggestion
	return apps, count, err
```

File: internal/project/repository/view/model/application.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+		OIDCPostLogoutRedirectUris: app.OIDCPostLogoutRedirectUris,
+	}
+}
+
+func OIDCResponseTypesFromModel(oidctypes []model.OIDCResponseType) []int64 {
+	result := make([]int64, 0)
+	for _, t := range oidctypes {
+		result = append(result, int64(t))
+	}
+	return result
```
Comment (adlerhurst) [inline internal/project/repository/view/model/application.go:67]:
```suggestion
	result := make([]int64, len(oidctypes))
	for i, t := range oidctypes {
		result[i] = int64(t)
	}
	return result
```

File: internal/project/repository/view/model/application.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+func OIDCGrantTypesFromModel(granttypes []model.OIDCGrantType) []int64 {
+	result := make([]int64, 0)
+	for _, t := range granttypes {
+		result = append(result, int64(t))
+	}
+	return result
```
Comment (adlerhurst) [inline internal/project/repository/view/model/application.go:75]:
```suggestion
	result := make([]int64, len(granttypes))
	for i, t := range granttypes {
		result[i] = int64(t)
	}
	return result
```

File: internal/project/repository/view/model/application.go [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+		OIDCPostLogoutRedirectUris: app.OIDCPostLogoutRedirectUris,
+	}
+}
+
+func OIDCResponseTypesToModel(oidctypes []int64) []model.OIDCResponseType {
+	result := make([]model.OIDCResponseType, 0)
+	for _, t := range oidctypes {
+		result = append(result, model.OIDCResponseType(t))
+	}
+	return result
```
Comment (adlerhurst) [inline internal/project/repository/view/model/application.go:104]:
```suggestion
	result := make([]model.OIDCResponseType, len(oidctypes))
	for i, t := range oidctypes {
		result[i] = model.OIDCResponseType(t)
	}
	return result
```

File: internal/project/repository/view/model/application.go [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+func OIDCGrantTypesToModel(granttypes []int64) []model.OIDCGrantType {
+	result := make([]model.OIDCGrantType, 0)
+	for _, t := range granttypes {
+		result = append(result, model.OIDCGrantType(t))
+	}
+	return result
```
Comment (adlerhurst) [inline internal/project/repository/view/model/application.go:112]:
```suggestion
	result := make([]model.OIDCGrantType, len(granttypes))
	for i, t := range granttypes {
		result[i] = model.OIDCGrantType(t)
	}
	return result
```

File: internal/project/repository/view/model/application.go [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+func ApplicationViewsToModel(roles []*ApplicationView) []*model.ApplicationView {
+	result := make([]*model.ApplicationView, 0)
+	for _, r := range roles {
+		result = append(result, ApplicationViewToModel(r))
+	}
+	return result
```
Comment (adlerhurst) [inline internal/project/repository/view/model/application.go:120]:
```suggestion
	result := make([]*model.ApplicationView, len(roles))
	for i, r := range roles {
		result[i] = ApplicationViewToModel(r)
	}
	return result
```

File: internal/project/repository/view/model/application.go [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+func (a *ApplicationView) setRootData(event *models.Event) {
+	a.ProjectID = event.AggregateID
+}
+
+func (a *ApplicationView) SetData(event *models.Event) error {
+	if err := json.Unmarshal(event.Data, a); err != nil {
+		logging.Log("EVEN-lo9ds").WithError(err).Error("could not unmarshal event data")
+		return err
```
Comment (adlerhurst) [inline internal/project/repository/view/model/application.go:152]: caos err

File: internal/project/repository/view/model/application_query.go [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+
+func (req ApplicationSearchRequest) GetAsc() bool {
+	return req.Asc
+}
+
+func (req ApplicationSearchRequest) GetQueries() []view.SearchQuery {
+	result := make([]view.SearchQuery, 0)
+	for _, q := range req.Queries {
+		result = append(result, ApplicationSearchQuery{Key: q.Key, Value: q.Value, Method: q.Method})
+	}
```
Comment (adlerhurst) [inline internal/project/repository/view/model/application_query.go:36]:
```suggestion
	result := make([]view.SearchQuery, len(req.Queries))
	for i, q := range req.Queries {
		result[i] = ApplicationSearchQuery{Key: q.Key, Value: q.Value, Method: q.Method}
	}
```

File: internal/project/repository/view/model/granted_project.go [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+		Sequence:      project.Sequence,
+	}
+}
+
+func GrantedProjectsToModel(projects []*GrantedProjectView) []*model.GrantedProjectView {
+	result := make([]*model.GrantedProjectView, 0)
+	for _, p := range projects {
+		result = append(result, GrantedProjectToModel(p))
+	}
+	return result
```
Comment (adlerhurst) [inline internal/project/repository/view/model/granted_project.go:81]:
```suggestion
	result := make([]*model.GrantedProjectView, len(projects))
	for i, p := range projects {
		result[i] = GrantedProjectToModel(p)
	}
	return result
```

File: internal/project/repository/view/model/granted_project.go [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+func (p *GrantedProjectView) setRootData(event *models.Event) {
+	p.ProjectID = event.AggregateID
+	p.OrgID = event.ResourceOwner
+	p.ResourceOwner = event.ResourceOwner
+}
+
+func (p *GrantedProjectView) setData(event *models.Event) error {
+	if err := json.Unmarshal(event.Data, p); err != nil {
+		logging.Log("EVEN-dlo92").WithError(err).Error("could not unmarshal event data")
+		return err
```
Comment (adlerhurst) [inline internal/project/repository/view/model/granted_project.go:123]: caos err

File: internal/project/repository/view/model/granted_project.go [hunk 13]
```diff
@@ -0,0 +1,148 @@
...
+	}
+	p.GrantID = grant.GrantID
+	p.GrantedRoleKeys = grant.RoleKeys
+	return nil
+}
+
+func (p *ProjectGrant) SetData(event *models.Event) error {
+	if err := json.Unmarshal(event.Data, p); err != nil {
+		logging.Log("EVEN-dlo92").WithError(err).Error("could not unmarshal event data")
+		return err
```
Comment (adlerhurst) [inline internal/project/repository/view/model/granted_project.go:145]: caos err

File: internal/project/repository/view/model/granted_project_query.go [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+func (req GrantedProjectSearchRequest) GetAsc() bool {
+	return req.Asc
+}
+
+func (req GrantedProjectSearchRequest) GetQueries() []view.SearchQuery {
+	result := make([]view.SearchQuery, 0)
+	for _, q := range req.Queries {
+		result = append(result, GrantedProjectSearchQuery{Key: q.Key, Value: q.Value, Method: q.Method})
+	}
```
Comment (adlerhurst) [inline internal/project/repository/view/model/granted_project_query.go:36]:
```suggestion
	result := make([]view.SearchQuery, len(req.Queries))
	for i, q := range req.Queries {
		result[i] = GrantedProjectSearchQuery{Key: q.Key, Value: q.Value, Method: q.Method}
	}
```

File: internal/project/repository/view/model/project_grant_member.go [hunk 15]
```diff
@@ -0,0 +1,101 @@
...
+		CreationDate: member.CreationDate,
+		ChangeDate:   member.ChangeDate,
+	}
+}
+
+func ProjectGrantMembersToModel(roles []*ProjectGrantMemberView) []*model.ProjectGrantMemberView {
+	result := make([]*model.ProjectGrantMemberView, 0)
+	for _, r := range roles {
+		result = append(result, ProjectGrantMemberToModel(r))
+	}
```
Comment (adlerhurst) [inline internal/project/repository/view/model/project_grant_member.go:73]:
```suggestion
	result := make([]*model.ProjectGrantMemberView, len(roles))
	for i, r := range roles {
		result[i] = ProjectGrantMemberToModel(r)
	}
```

File: internal/project/repository/view/model/project_grant_member.go [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+}
+
+func (r *ProjectGrantMemberView) setRootData(event *models.Event) {
+	r.ProjectID = event.AggregateID
+}
+
+func (r *ProjectGrantMemberView) SetData(event *models.Event) error {
+	if err := json.Unmarshal(event.Data, r); err != nil {
+		logging.Log("EVEN-slo9s").WithError(err).Error("could not unmarshal event data")
+		return err
```
Comment (adlerhurst) [inline internal/project/repository/view/model/project_grant_member.go:98]: caos err

File: internal/project/repository/view/model/project_grant_member_query.go [hunk 17]
```diff
@@ -0,0 +1,69 @@
...
+
+func (req ProjectGrantMemberSearchRequest) GetAsc() bool {
+	return req.Asc
+}
+
+func (req ProjectGrantMemberSearchRequest) GetQueries() []view.SearchQuery {
+	result := make([]view.SearchQuery, 0)
+	for _, q := range req.Queries {
+		result = append(result, ProjectGrantMemberSearchQuery{Key: q.Key, Value: q.Value, Method: q.Method})
+	}
```
Comment (adlerhurst) [inline internal/project/repository/view/model/project_grant_member_query.go:36]:
```suggestion
	result := make([]view.SearchQuery, len(req.Queries))
	for i, q := range req.Queries {
		result[i] = ProjectGrantMemberSearchQuery{Key: q.Key, Value: q.Value, Method: q.Method}
	}
```

File: internal/project/repository/view/model/project_member.go [hunk 18]
```diff
@@ -0,0 +1,98 @@
...
+		CreationDate: member.CreationDate,
+		ChangeDate:   member.ChangeDate,
+	}
+}
+
+func ProjectMembersToModel(roles []*ProjectMemberView) []*model.ProjectMemberView {
+	result := make([]*model.ProjectMemberView, 0)
+	for _, r := range roles {
+		result = append(result, ProjectMemberToModel(r))
+	}
```
Comment (adlerhurst) [inline internal/project/repository/view/model/project_member.go:70]:
```suggestion
	result := make([]*model.ProjectMemberView, len(roles))
	for i, r := range roles {
		result[i] = ProjectMemberToModel(r)
	}
```

File: internal/project/repository/view/model/project_member.go [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+}
+
+func (r *ProjectMemberView) setRootData(event *models.Event) {
+	r.ProjectID = event.AggregateID
+}
+
+func (r *ProjectMemberView) SetData(event *models.Event) error {
+	if err := json.Unmarshal(event.Data, r); err != nil {
+		logging.Log("EVEN-slo9s").WithError(err).Error("could not unmarshal event data")
+		return err
```
Comment (adlerhurst) [inline internal/project/repository/view/model/project_member.go:95]: caos err

File: internal/project/repository/view/model/project_member_query.go [hunk 20]
```diff
@@ -0,0 +1,69 @@
...
+
+func (req ProjectMemberSearchRequest) GetAsc() bool {
+	return req.Asc
+}
+
+func (req ProjectMemberSearchRequest) GetQueries() []view.SearchQuery {
+	result := make([]view.SearchQuery, 0)
+	for _, q := range req.Queries {
+		result = append(result, ProjectMemberSearchQuery{Key: q.Key, Value: q.Value, Method: q.Method})
+	}
```
Comment (adlerhurst) [inline internal/project/repository/view/model/project_member_query.go:36]:
```suggestion
	result := make([]view.SearchQuery, len(req.Queries))
	for i, q := range req.Queries {
		result[i] = ProjectMemberSearchQuery{Key: q.Key, Value: q.Value, Method: q.Method}
	}
```

File: internal/project/repository/view/model/project_role.go [hunk 21]
```diff
@@ -0,0 +1,90 @@
...
+		Sequence:      role.Sequence,
+		CreationDate:  role.CreationDate,
+	}
+}
+
+func ProjectRolesToModel(roles []*ProjectRoleView) []*model.ProjectRoleView {
+	result := make([]*model.ProjectRoleView, 0)
+	for _, r := range roles {
+		result = append(result, ProjectRoleToModel(r))
+	}
```
Comment (adlerhurst) [inline internal/project/repository/view/model/project_role.go:61]:
```suggestion
	result := make([]*model.ProjectRoleView, len(roles))
	for i, r := range roles {
		result[i] = ProjectRoleToModel(r)
	}
```

File: internal/project/repository/view/model/project_role.go [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+func (r *ProjectRoleView) setRootData(event *models.Event) {
+	r.ProjectID = event.AggregateID
+	r.OrgID = event.ResourceOwner
+	r.ResourceOwner = event.ResourceOwner
+}
+
+func (r *ProjectRoleView) SetData(event *models.Event) error {
+	if err := json.Unmarshal(event.Data, r); err != nil {
+		logging.Log("EVEN-slo9s").WithError(err).Error("could not unmarshal event data")
+		return err
```
Comment (adlerhurst) [inline internal/project/repository/view/model/project_role.go:87]: caos err

File: internal/project/repository/view/model/project_role_query.go [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+
+func (req ProjectRoleSearchRequest) GetAsc() bool {
+	return req.Asc
+}
+
+func (req ProjectRoleSearchRequest) GetQueries() []view.SearchQuery {
+	result := make([]view.SearchQuery, 0)
+	for _, q := range req.Queries {
+		result = append(result, ProjectRoleSearchQuery{Key: q.Key, Value: q.Value, Method: q.Method})
+	}
```
Comment (adlerhurst) [inline internal/project/repository/view/model/project_role_query.go:36]:
```suggestion
	result := make([]view.SearchQuery, len(req.Queries))
	for i, q := range req.Queries {
		result[i] = ProjectRoleSearchQuery{Key: q.Key, Value: q.Value, Method: q.Method}
	}
```

File: pkg/management/api/grpc/application_converter.go [hunk 24]
```diff
@@ -90,6 +102,79 @@ func oidcConfigUpdateToModel(app *OIDCConfigUpdate) *proj_model.OIDCConfig {
...
+		Limit:   request.Limit,
+		Queries: applicationSearchQueriesToModel(request.Queries),
+	}
+}
+
+func applicationSearchQueriesToModel(queries []*ApplicationSearchQuery) []*proj_model.ApplicationSearchQuery {
+	converted := make([]*proj_model.ApplicationSearchQuery, 0)
+	for _, q := range queries {
+		converted = append(converted, applicationSearchQueryToModel(q))
+	}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/application_converter.go:117]:
```suggestion
	converted := make([]*proj_model.ApplicationSearchQuery, len(queries))
	for i, q := range queries {
		converted[i] = applicationSearchQueryToModel(q)
	}
```

File: pkg/management/api/grpc/application_converter.go [hunk 25]
```diff
@@ -90,6 +102,79 @@ func oidcConfigUpdateToModel(app *OIDCConfigUpdate) *proj_model.OIDCConfig {
...
+		TotalResult: response.TotalResult,
+		Result:      applicationViewsFromModel(response.Result),
+	}
+}
+
+func applicationViewsFromModel(apps []*proj_model.ApplicationView) []*ApplicationView {
+	converted := make([]*ApplicationView, 0)
+	for _, q := range apps {
+		converted = append(converted, applicationViewFromModel(q))
+	}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/application_converter.go:151]:
```suggestion
	converted := make([]*ApplicationView, len(apps))
	for i, app := range apps {
		converted[i] = applicationViewFromModel(app)
	}
```

File: pkg/management/api/grpc/project_converter.go [hunk 26]
```diff
@@ -24,6 +24,76 @@ func projectFromModel(project *proj_model.Project) *Project {
...
+		TotalResult: response.TotalResult,
+		Result:      grantedProjectsFromModel(response.Result),
+	}
+}
+
+func grantedProjectsFromModel(projects []*proj_model.GrantedProjectView) []*GrantedProject {
+	converted := make([]*GrantedProject, 0)
+	for _, q := range projects {
+		converted = append(converted, grantedProjectFromModel(q))
+	}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/project_converter.go:40]:
```suggestion
	converted := make([]*GrantedProject, len(projects))
	for i, project := range projects {
		converted[i] = grantedProjectFromModel(project)
	}
```

File: pkg/management/api/grpc/project_converter.go [hunk 27]
```diff
@@ -24,6 +24,76 @@ func projectFromModel(project *proj_model.Project) *Project {
...
+		TotalResult: response.TotalResult,
+		Result:      projectRoleViewsFromModel(response.Result),
+	}
+}
+
+func projectRoleViewsFromModel(roles []*proj_model.ProjectRoleView) []*ProjectRoleView {
+	converted := make([]*ProjectRoleView, 0)
+	for _, q := range roles {
+		converted = append(converted, projectRoleViewFromModel(q))
+	}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/project_converter.go:79]:
```suggestion
	converted := make([]*ProjectRoleView, len(roles))
	for i, role := range roles {
		converted[i] = projectRoleViewFromModel(role)
	}
```

File: pkg/management/api/grpc/project_converter.go [hunk 28]
```diff
@@ -82,3 +163,71 @@ func projectRoleChangeToModel(role *ProjectRoleChange) *proj_model.ProjectRole {
...
+		Limit:   project.Limit,
+		Queries: grantedProjectSearchQueriesToModel(project.Queries),
+	}
+}
+
+func grantedProjectSearchQueriesToModel(queries []*GrantedProjectSearchQuery) []*proj_model.GrantedProjectSearchQuery {
+	converted := make([]*proj_model.GrantedProjectSearchQuery, 0)
+	for _, q := range queries {
+		converted = append(converted, grantedProjectSearchQueryToModel(q))
+	}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/project_converter.go:179]:
```suggestion
	converted := make([]*proj_model.GrantedProjectSearchQuery, len(queries))
	for i, q := range queries {
		converted[i] = grantedProjectSearchQueryToModel(q)
	}
```

File: pkg/management/api/grpc/project_converter.go [hunk 29]
```diff
@@ -82,3 +163,71 @@ func projectRoleChangeToModel(role *ProjectRoleChange) *proj_model.ProjectRole {
...
+		Limit:   role.Limit,
+		Queries: projectRoleSearchQueriesToModel(role.Queries),
+	}
+}
+
+func projectRoleSearchQueriesToModel(queries []*ProjectRoleSearchQuery) []*proj_model.ProjectRoleSearchQuery {
+	converted := make([]*proj_model.ProjectRoleSearchQuery, 0)
+	for _, q := range queries {
+		converted = append(converted, projectRoleSearchQueryToModel(q))
+	}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/project_converter.go:212]:
```suggestion
	converted := make([]*proj_model.ProjectRoleSearchQuery, len(queries))
	for i, q := range queries {
		converted[i] = projectRoleSearchQueryToModel(q)
	}
```

File: pkg/management/api/grpc/project_grant_converter.go [hunk 30]
```diff
@@ -45,6 +46,61 @@ func projectGrantUpdateToModel(grant *ProjectGrantUpdate) *proj_model.ProjectGra
...
+		TotalResult: response.TotalResult,
+		Result:      projectGrantsFromGrantedProjectModel(response.Result),
+	}
+}
+
+func projectGrantsFromGrantedProjectModel(projects []*proj_model.GrantedProjectView) []*ProjectGrantView {
+	converted := make([]*ProjectGrantView, 0)
+	for _, q := range projects {
+		converted = append(converted, projectGrantFromGrantedProjectModel(q))
+	}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/project_grant_converter.go:79]:
```suggestion
	converted := make([]*ProjectGrantView, len(projects))
	for i, project := range projects {
		converted = projectGrantFromGrantedProjectModel(project)
	}
```

File: pkg/management/api/grpc/project_grant_member_converter.go [hunk 31]
```diff
@@ -44,3 +44,79 @@ func projectGrantMemberChangeToModel(member *ProjectGrantMemberChange) *proj_mod
...
+		Limit:   role.Limit,
+		Queries: projectGrantMemberSearchQueriesToModel(role.Queries),
+	}
+}
+
+func projectGrantMemberSearchQueriesToModel(queries []*ProjectGrantMemberSearchQuery) []*proj_model.ProjectGrantMemberSearchQuery {
+	converted := make([]*proj_model.ProjectGrantMemberSearchQuery, 0)
+	for _, q := range queries {
+		converted = append(converted, projectGrantMemberSearchQueryToModel(q))
+	}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/project_grant_member_converter.go:60]:
```suggestion
	converted := make([]*proj_model.ProjectGrantMemberSearchQuery, len(queries))
	for i, q := range queries {
		converted[i] = projectGrantMemberSearchQueryToModel(q)
	}
```

File: pkg/management/api/grpc/project_grant_member_converter.go [hunk 32]
```diff
@@ -44,3 +44,79 @@ func projectGrantMemberChangeToModel(member *ProjectGrantMemberChange) *proj_mod
...
+		TotalResult: response.TotalResult,
+		Result:      projectGrantMemberViewsFromModel(response.Result),
+	}
+}
+
+func projectGrantMemberViewsFromModel(roles []*proj_model.ProjectGrantMemberView) []*ProjectGrantMemberView {
+	converted := make([]*ProjectGrantMemberView, 0)
+	for _, q := range roles {
+		converted = append(converted, projectGrantMemberViewFromModel(q))
+	}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/project_grant_member_converter.go:100]:
```suggestion
	converted := make([]*ProjectGrantMemberView, len(roles))
	for i, role := range roles {
		converted[i] = projectGrantMemberViewFromModel(role)
	}
```

File: pkg/management/api/grpc/project_member_converter.go [hunk 33]
```diff
@@ -42,3 +42,79 @@ func projectMemberChangeToModel(member *ProjectMemberChange) *proj_model.Project
...
+		Limit:   role.Limit,
+		Queries: projectMemberSearchQueriesToModel(role.Queries),
+	}
+}
+
+func projectMemberSearchQueriesToModel(queries []*ProjectMemberSearchQuery) []*proj_model.ProjectMemberSearchQuery {
+	converted := make([]*proj_model.ProjectMemberSearchQuery, 0)
+	for _, q := range queries {
+		converted = append(converted, projectMemberSearchQueryToModel(q))
+	}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/project_member_converter.go:58]:
```suggestion
	converted := make([]*proj_model.ProjectMemberSearchQuery, len(queries))
	for i, q := range queries {
		converted[i] = projectMemberSearchQueryToModel(q)
	}
```

File: pkg/management/api/grpc/project_member_converter.go [hunk 34]
```diff
@@ -42,3 +42,79 @@ func projectMemberChangeToModel(member *ProjectMemberChange) *proj_model.Project
...
+		TotalResult: response.TotalResult,
+		Result:      projectMemberViewsFromModel(response.Result),
+	}
+}
+
+func projectMemberViewsFromModel(roles []*proj_model.ProjectMemberView) []*ProjectMemberView {
+	converted := make([]*ProjectMemberView, 0)
+	for _, q := range roles {
+		converted = append(converted, projectMemberViewFromModel(q))
+	}
```
Comment (adlerhurst) [inline pkg/management/api/grpc/project_member_converter.go:98]:
```suggestion
func projectMemberViewsFromModel(members []*proj_model.ProjectMemberView) []*ProjectMemberView {
	converted := make([]*ProjectMemberView, len(members))
	for i, member := range members {
		converted[i] = projectMemberViewFromModel(member)
	}
```

**Pull request merged**